cmd/soroban-rpc/internal/db/migrations/02_transactions.sql

```diff
@@ -0,0 +1,16 @@
+-- +migrate Up
+
+-- index to find transactions in ledgers by hash
+CREATE TABLE transactions (
+    hash NCHAR(64) PRIMARY KEY,
```

Comment:
if you store the hash in binary instead of the hexadecimal form you only require 32 bytes

Reply:
I wanted to do this at first but I couldn't figure out how to make `BLOB` play nicely with `[]byte` when calling `.Values(...)`. I think it kept wanting to expand the slice into individual components... but I tried again in f265e19f0c8a64808f5c0b85caa31fbbe983b2a5 and it worked this time for some reason lol.

A downside is that it makes manual inspection of the database harder (since `select hash from transactions limit 1;` gives you nonsense that you can't forward to a `curl getTransaction` RPC call)

Reply:
you can use the `hex()` function in the sql query to render the BLOB as a hex string:

https://evanhahn.com/how-to-format-sqlite-blob-columns-as-hex/

Reply:
protip, thanks!